server/src/main/java/org/elasticsearch/health/HealthIndicatorService.java

```diff
@@ -55,4 +55,14 @@ default HealthIndicatorResult createIndicator(
             .collect(Collectors.toList());
         return new HealthIndicatorResult(name(), status, symptom, details, impactsList, diagnosisList);
     }
+
+    /**
+     * A preflight indicator is an indicator that is run first and represents a serious cascading health problem. For example, the
+     * `stable_master` health indicator is a preflight indicator. When it is red it means that we do not have a formed cluster so running
+     * further diagnostics is not valuable.
```

Comment:
I believe this is not entirely correct - a node has witnessed too many maser nodes which could mean there are missing nodes, or a discovery problem, or that the node itself has problems joining the elected master. 
Furthermore, the reason all the other indicators turn to `unknown` is that the local cluster state is likely not-up-to date enough for us to make health decisions off of it.

Reply:
Thank you @andreidan, very good point I will adjust it.